CraftBukkit-Patches/0024-Server-difficulty-setting.patch

```diff
@@ -0,0 +1,94 @@
+From 076ef05d9ae6dad129b36a64984727417fccb4c7 Mon Sep 17 00:00:00 2001
+From: mintoyatsu <53593903+mintoyatsu@users.noreply.github.com>
+Date: Tue, 10 Mar 2020 03:40:34 -0700
+Subject: [PATCH] Server difficulty setting
+
+Allows difficulty selection using server.properties. The "difficulty"
+property has four values: 0 (Peaceful), 1 (Easy), 2 (Normal), 3 (Hard).
+"spawn-monsters" can still be used to disable monster spawning
+regardless of difficulty.
+
+diff --git a/src/main/java/net/minecraft/server/EntityHuman.java b/src/main/java/net/minecraft/server/EntityHuman.java
+index a57fbf77..733c742f 100644
+--- a/src/main/java/net/minecraft/server/EntityHuman.java
++++ b/src/main/java/net/minecraft/server/EntityHuman.java
+@@ -175,8 +175,8 @@ public abstract class EntityHuman extends EntityLiving {
+     }
+ 
+     public void v() {
+-        // CraftBukkit - spawnMonsters -> allowMonsters
+-        if (!this.world.allowMonsters && this.health < 20 && this.ticksLived % 20 * 12 == 0) {
++        if (this.world.spawnMonsters == 0 && this.health < 20 && this.ticksLived % 20 * 12 == 0) {
```

Comment:
Please comment this change from CraftBukkit

Reply:
This is indeed intentional. When b1.8 was released the Bukkit team reverted this line because it gunked up health regen on difficulty=0. Checking the decompiled vanilla server in MCP shows that this line is the same, thus no comment is needed.

See: [CraftBukkit commit 0a52c7f3346833fd568e020d3f1b918b792b09f4](https://github.com/OvercastNetwork/CraftBukkit/commit/0a52c7f3346833fd568e020d3f1b918b792b09f4)

Reply:
Sounds good.